regression_tests/compute_mse.jl

```diff
@@ -29,7 +29,7 @@ function regression_test(; job_id, reference_mse, ds_filename_computed, varname)
     if haskey(ENV, "BUILDKITE_COMMIT")
         path = self_reference_or_path()
         path == :self_reference && return reference_mse
-        ds_filename_reference = joinpath(path, ds_filename_computed)
+        ds_filename_reference = joinpath(path, job_id, "prog_state.nc")
```

Comment:
Is this necessary? How is this change used?

Reply:
This is necessary in order to run multiple pipeline examples that compare against the same reference data, like [here](https://github.com/CliMA/ClimaAtmos.jl/blob/fe3b9d47c74b15304723c00a32561c534ca66f0d/.buildkite/pipeline.yml#L313-L323).

Reply:
Ok, I think this is fine to try out